src/Features/Core/Portable/EmbeddedLanguages/Json/JsonParser.StrictSyntaxChecker.cs

```diff
@@ -0,0 +1,338 @@
+                    }
+                }
+
+                return CheckProperSeparation(sequence) ?? CheckChildren(node);
+            }
+
+            private static EmbeddedDiagnostic? CheckArray(JsonArrayNode node)
+            {
+                foreach (var child in node.Sequence)
+                {
+                    if (child.Kind == JsonKind.Property)
+                    {
+                        return new EmbeddedDiagnostic(
+                            FeaturesResources.Properties_not_allowed_in_an_array,
+                            ((JsonPropertyNode)child).ColonToken.GetSpan());
+                    }
+                }
+
+                return CheckProperSeparation(node.Sequence) ?? CheckChildren(node);
+            }
+
+            private static EmbeddedDiagnostic? CheckProperSeparation(ImmutableArray<JsonValueNode> sequence)
+            {
+                // Ensure that this sequence is actually a separated list.
+                for (int i = 0, n = sequence.Length; i < n; i++)
+                {
+                    var child = sequence[i];
+                    if (i % 2 == 0)
+                    {
+                        if (child.Kind == JsonKind.CommaValue)
+                        {
+                            return new EmbeddedDiagnostic(
+                                string.Format(FeaturesResources._0_unexpected, ","),
+                                child.GetSpan());
+                        }
+                    }
+                    else
+                    {
+                        if (child.Kind != JsonKind.CommaValue)
+                        {
+                            return new EmbeddedDiagnostic(
+                                string.Format(FeaturesResources._0_expected, ","),
+                                GetFirstToken(child).GetSpan());
+                        }
+                    }
+                }
+
+                if (sequence.Length != 0 &&
+                    sequence.Length % 2 == 0)
+                {
+                    var lastChild = sequence[^1];
+                    return new EmbeddedDiagnostic(
+                        FeaturesResources.Trailing_comma_not_allowed,
+                        lastChild.GetSpan());
+                }
+
+                return null;
+            }
+
+            private static EmbeddedDiagnostic? CheckProperty(JsonPropertyNode node)
+            {
+                if (node.NameToken.Kind != JsonKind.StringToken)
+                {
+                    return new EmbeddedDiagnostic(
+                        FeaturesResources.Property_name_must_be_a_string,
+                        node.NameToken.GetSpan());
+                }
+
+                if (node.Value.Kind == JsonKind.CommaValue)
+                {
+                    return new EmbeddedDiagnostic(
+                        FeaturesResources.Value_required,
+                        new TextSpan(node.ColonToken.VirtualChars[0].Span.End, 0));
+                }
```

Comment:
What about something like `{ a: }`, we won't have a comma there?

Reply:
You get the error:

```
<Diagnostics>
  <Diagnostic Message=""'}' unexpected"" Start=""16"" Length=""1"" />
</Diagnostics>
```

I added a test.  We don't even get to the 'syntax-checking' pass becuse the core parsing failed due to invalid code.